backend/src/graphql/resolver/BalanceResolver.ts

```diff
@@ -5,36 +5,74 @@ import { Resolver, Query, Ctx, Authorized } from 'type-graphql'
 import { Balance } from '@model/Balance'
 import { calculateDecay } from '@/util/decay'
 import { RIGHTS } from '@/auth/RIGHTS'
-import { Transaction } from '@entity/Transaction'
+import { Transaction as dbTransaction } from '@entity/Transaction'
 import Decimal from 'decimal.js-light'
+import { GdtResolver } from './GdtResolver'
+import { TransactionLink as dbTransactionLink } from '@entity/TransactionLink'
+import { MoreThan, getCustomRepository } from '@dbTools/typeorm'
+import { TransactionLinkRepository } from '@repository/TransactionLink'
 
 @Resolver()
 export class BalanceResolver {
   @Authorized([RIGHTS.BALANCE])
   @Query(() => Balance)
   async balance(@Ctx() context: any): Promise<Balance> {
-    // load user and balance
     const { user } = context
     const now = new Date()
 
-    const lastTransaction = await Transaction.findOne(
-      { userId: user.id },
-      { order: { balanceDate: 'DESC' } },
-    )
+    const gdtResolver = new GdtResolver()
+    const balanceGDT = await gdtResolver.gdtBalance(context)
+
+    const lastTransaction = context.lastTransaction
+      ? context.lastTransaction
+      : await dbTransaction.findOne({ userId: user.id }, { order: { balanceDate: 'DESC' } })
 
     // No balance found
     if (!lastTransaction) {
       return new Balance({
         balance: new Decimal(0),
         decay: new Decimal(0),
-        decay_date: now.toString(),
+        lastBookedBalance: new Decimal(0),
+        balanceGDT,
+        count: 0,
+        linkCount: 0,
       })
     }
 
+    const count =
+      context.count || context.count === 0
+        ? context.count
+        : await dbTransaction.count({ where: { userId: user.id } })
+    const linkCount =
+      context.linkCount || context.linkCount === 0
+        ? context.linkCount
+        : await dbTransactionLink.count({
+            where: {
+              userId: user.id,
+              redeemedAt: null,
+              validUntil: MoreThan(new Date()),
+            },
+          })
+
+    const transactionLinkRepository = getCustomRepository(TransactionLinkRepository)
+    const { sumHoldAvailableAmount } = context.sumHoldAvailableAmount
+      ? { sumHoldAvailableAmount: context.sumHoldAvailableAmount }
+      : await transactionLinkRepository.summary(user.id, now)
+
+    const calculatedDecay = calculateDecay(
+      lastTransaction.balance.minus(sumHoldAvailableAmount.toString()),
+      lastTransaction.balanceDate,
+      now,
+    )
+
     return new Balance({
-      balance: lastTransaction.balance,
-      decay: calculateDecay(lastTransaction.balance, lastTransaction.balanceDate, now).balance,
-      decay_date: now.toString(),
+      balance: calculatedDecay.balance.toDecimalPlaces(2, Decimal.ROUND_DOWN), // round towards zero
+      decay: calculatedDecay.decay.toDecimalPlaces(2, Decimal.ROUND_FLOOR), // round towards - infinity
```

Comment:
Why round towards -infinity and not +infinity? Do we not to round up the decay value?
![image](https://user-images.githubusercontent.com/1238238/160589621-ceff7b03-e564-4c4b-88dc-8ee65c963277.png)

Reply:
Decay is always negative. E.g. `- 1.2398`. Round towards `+ infinity` gives `-1.23`, round towards `- infinity` becomes `-1.24`, this is what I want.